pinot-tools/src/main/java/com/linkedin/pinot/tools/admin/command/MoveReplicaGroup.java

```diff
@@ -0,0 +1,440 @@
+        .getRecord().getMapFields();
+
+    System.out.println("Existing idealstate:");
+    printIdealState(idealStateMap);
+
+    PriorityQueue<SourceSegments> segmentsToMove = getSegmentsToMoveQueue(idealStateMap, srcHosts);
+    PriorityQueue<ServerInstance> destinationServerQueue = getDestinationServerQueue(idealStateMap, destinationServers);
+
+    Map<String, Map<String, String>> proposedIdealState =
+        computeNewIdealState(idealStateMap, segmentsToMove, destinationServerQueue, srcHosts);
+    System.out.println("Proposed idealstate:");
+    printIdealState(proposedIdealState);
+    printDestinationServerCounts(destinationServerQueue);
+    if (! exec) {
+      LOGGER.info("Run with -exec to apply this IdealState");
+      System.exit(0);
+    }
+
+    applyIdealState(proposedIdealState);
+    zkChanger.waitForStable(tableName);
+    return true;
+  }
+
+  private void printDestinationServerCounts(PriorityQueue<ServerInstance> destinationServerQueue) {
+    System.out.println("Number of segments per server: ");
+    for (ServerInstance instance : destinationServerQueue) {
+      System.out.println(instance.server + " : " + instance.segments);
+    }
+  }
+
+  private void printIdealState(Map<String, Map<String, String>> idealState)
+      throws JsonProcessingException {
+    String output = new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(idealState);
+    System.out.println(output);
+  }
+
+  private void applyIdealState(final Map<String, Map<String, String>> proposedIdealState) {
+    HelixHelper.updateIdealState(zkChanger.getHelixManager(), tableName, new Function<IdealState, IdealState>() {
+      @Nullable
+      @Override
+      public IdealState apply(@Nullable IdealState input) {
+        Map<String, Map<String, String>> existingMapField = input.getRecord().getMapFields();
+
+        for (Map.Entry<String, Map<String, String>> segmentEntry : proposedIdealState.entrySet()) {
+          existingMapField.put(segmentEntry.getKey(), segmentEntry.getValue());
+        }
+        return input;
+      }
+    }, RetryPolicies.exponentialBackoffRetryPolicy(5, 500L, 2.0f));
+  }
+
+  private Map<String, Map<String, String>> computeNewIdealState(Map<String, Map<String, String>> idealStateMap,
+      PriorityQueue<SourceSegments> segmentsToMove, PriorityQueue<ServerInstance> destinationServers,
+      List<String> srcHosts) {
+
+    Map<String, Map<String, String>> newIdealState = copyIdealState(idealStateMap);
+
+    for (int remapCount = 0; remapCount < maxSegmentsToMove && !segmentsToMove.isEmpty(); ++remapCount) {
+      String segment = segmentsToMove.poll().segment;
+      Map<String, String> existingMapping = newIdealState.get(segment);
+      String destinationServer = getDestinationServer(destinationServers, existingMapping);
+      if (destinationServer == null) {
+        throw new RuntimeException("No destination server for segment: " + segment);
```

Comment:
Instead of throwing an exception, do you want to continue and process other segemnts?